target/scripts/helpers/log.sh

```diff
@@ -116,4 +108,34 @@ function _log
   fi
 }
 
-function _log_with_date { _log "${1}" "$(date '+%Y-%m-%d %H:%M:%S')  ${2}" ; }
+# Like `_log` but adds a timestamp in front of the message.
+function _log_with_date
+{
+  _log "${1}" "$(date '+%Y-%m-%d %H:%M:%S')  ${2}"
+}
+
+# Get the value of the environment variable LOG_LEVEL if
+# it is set. Otherwise, try to query the common environment
+# variables file. If this does not yield a value either,
+# use the default log level.
+function _get_log_level_or_default
+{
+  if [[ -n ${LOG_LEVEL+set} ]]
+  then
+    printf '%s' "${LOG_LEVEL}"
```

Comment:
<details>
<summary>Resolved</summary>

Is it desired for this to work with invalid `LOG_LEVEL` when set? (eg: `LOG_LEVEL=whatever`)

Previously we had:

https://github.com/docker-mailserver/docker-mailserver/blob/e6f481b59f7f67a7f605f9d97fa6a5ee1de68840/target/scripts/helpers/log.sh#L62-L78

Where `( *       ) LEVEL_AS_INT=3 ;;` is effectively only for `info`, as any other value would then hit:

https://github.com/docker-mailserver/docker-mailserver/blob/e6f481b59f7f67a7f605f9d97fa6a5ee1de68840/target/scripts/helpers/log.sh#L103-L107

Which now is an `error` type log.

</details>

---

**EDIT:** Ok, I grok the flow now :sweat_smile: 

- I think `/etc/dms-settings` will still only ever be hit when `LOG_LEVEL` isn't set, thus almost always is `info` (_unless overriding the ENV for commands with `docker exec` or modifying `/etc/dms-settings` for some reason_).
- We don't handle bad `LOG_LEVEL` early (_other than making mistakes more apparent via the `error` log fallback in this PR_), which should always be due to `LOG_LEVEL` being set wrong.
- Due to those two niche scenarios, `log.sh` handles this internally, instead of `variables.sh` at startup.

Reply:
I guess you provided the solution to your own question (?) :D

Reply:
Yeah, I eventually figured it out :sweat_smile: :+1: